diffrax/_term.py

```diff
@@ -335,9 +357,18 @@ class ControlTerm(_AbstractControlTerm[_VF, _Control]):
     """
 
     def prod(self, vf: _VF, control: _Control) -> Y:
+        if isinstance(vf, lx.AbstractLinearOperator):
+            return jtu.tree_map(
+                lambda _vf, _control: _vf.mv(_control),
+                vf,
+                control,
+                is_leaf=lambda x: eqx.is_array(x)
```

Comment:
I think the `eqx.is_array` bit is unnecessary -- anything matching this will already be a leaf.

Reply:
Actually also, show we have a `tree_map` here or not? The surrounding check is already `isinstance(vf, lx.AbstractLinearOperator)`, so this tree-map is guaranteed to be trivial.

Off the top of my head I don't know if I have an opinion on whether `vf` should be either a `lx.AbstractLinearOperator` or a `PyTree[lx.AbstractLinearOperator]`. WDYT?

Reply:
True, I removed the `is_array`. Also true that the if/tree-map was unnecessary. I updated the if statement. 

Unless my understanding of lineax is wrong (which it could be, I'm new to that package), a PyTree[operator] seems like the way to go. Diffrax allows arbitrary pytrees as control, so the output of the vector field could previously be any pytree of arrays (that are just element wise multiplied). My understanding of how to do this in lineax was just to make a pytree of operators and treemap over them. If this is true, then I think PyTree[operator] is definitely the way to go, since pytree based controls are important. If there is a different way to do things in lineax then perhaps there are other approaches to consider here.

Reply:
So both approaches are equally expressive: you could write a linear operator that consumes a pytree and does whatever you like with it.

As such, doing a tree-map here is equivalent to making a block-diagonal operator.

Noodling it over, I think I'm probably leaning towards having just a single operator. This pushes more of the abstraction into Lineax (where it's easier to reason about I think), and means we can skip the potentially error-prone process of appropriately broadcasting PyTrees etc here. I always have to scrutinise tree-map code 10x as much as any other code!

Reply:
I see, so for this sort of block diagonal approach, the user should create a custom operator that inherits and implement .mv using this sort of tree map. But in diffrax itself, the code doesn't know about the pytree just calls .mv?

Reply:
Yup, exactly!

Reply:
Ok, I changed it